Tests/WitsmlExplorer.Api.Tests/Workers/LogWorkerToolsTests.cs

```diff
@@ -0,0 +1,189 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.Extensions.Logging;
+
+using Moq;
+
+using Serilog;
+
+using Witsml;
+using Witsml.Data;
+using Witsml.ServiceReference;
+
+using WitsmlExplorer.Api.Jobs;
+using WitsmlExplorer.Api.Models;
+using WitsmlExplorer.Api.Services;
+using WitsmlExplorer.Api.Workers;
+using WitsmlExplorer.Api.Workers.Create;
+
+using Xunit;
+
+namespace WitsmlExplorer.Api.Tests.Workers
+{
+    public class LogWorkerToolsTests
+    {
+        private const string LogUid = "51bb71c2-5e6f-4e15-ae5f-0fbc866bdad6";
+        private const string LogName = "Test log";
+        private const string WellUid = "welluid";
+        private const string WellName = "wellname";
+        private const string WellboreUid = "wellboreuid";
+        private const string WellboreName = "wellborename";
+        private readonly Mock<IWitsmlClient> _witsmlClient;
+        private readonly CreateObjectOnWellboreWorker _worker;
+
+        public LogWorkerToolsTests()
+        {
+            Mock<IWitsmlClientProvider> witsmlClientProvider = new();
+            _witsmlClient = new Mock<IWitsmlClient>();
+            witsmlClientProvider.Setup(provider => provider.GetClient()).Returns(_witsmlClient.Object);
+            ILoggerFactory loggerFactory = new LoggerFactory();
+            loggerFactory.AddSerilog(Log.Logger);
+            ILogger<CreateObjectOnWellboreJob> logger = loggerFactory.CreateLogger<CreateObjectOnWellboreJob>();
+            _worker = new CreateObjectOnWellboreWorker(logger, witsmlClientProvider.Object);
+        }
+
+        [Fact]
+        public async Task GetLogTest_OK()
```

Comment:
Instead of using _ReturnedWitsmlLog()_, it's easier to use _CreateLog()_ along with _AsItemInWitsmlList()_ as it makes the expected log more available. Then you can also use it for the properties instead of adding hard-coded expected values. Same goes for the other tests.

```
public async Task GetLogTest_OK()
{
    WitsmlLog expectedLog = CreateLog();
    CreateObjectOnWellboreJob job = CreateJobTemplate(WitsmlLog.WITSML_INDEX_TYPE_MD);
    _witsmlClient.Setup(client =>
        client.GetFromStoreAsync(It.IsAny<WitsmlLogs>(), It.Is<OptionsIn>((ops) => ops.ReturnElements == ReturnElements.HeaderOnly))).ReturnsAsync(expectedLog.AsItemInWitsmlList());

    var log = await LogWorkerTools.GetLog(_witsmlClient.Object, job.Object, ReturnElements.HeaderOnly);
    Assert.Equal(expectedLog.Uid, log.Uid);
    Assert.Equal(expectedLog.EndDateTimeIndex, log.EndDateTimeIndex);
    Assert.Equal(expectedLog.StartDateTimeIndex, log.StartDateTimeIndex);
    Assert.Equal(expectedLog.LogCurveInfo.Count, log.LogCurveInfo.Count);
}
```

Reply:
done